### PR TITLE
DPDK Backend: Fix verify statement support and few other minor issues

### PR DESCRIPTION
**Pull request by usha1830**
This PR fixes the following issues:

1) Handled boolean literals in verify statement
2) Moving error id from verify method to standard metadata for parser error
3) Emitting missing Label before drop instruction at the end.
4) Change warning type for mismatched header fields
5) Change cast to mov instruction as it is not a valid instruction.

### CODE REVIEW

File: backends/dpdk/dpdkHelpers.cpp [hunk 1]
```diff
@@ -619,11 +628,20 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
                 ::error("%1%: must be one of the existing errors", s);
             auto error_id = structure->error_map.at(error->expression->to<IR::Member>()->member);
             auto end_label = Util::printf_format("label_%dend", next_label_id++);
-            add_instr(new IR::DpdkJmpEqualStatement(
-                        end_label,
-                        condition->expression, new IR::BoolLiteral(false)));
-            add_instr(new IR::DpdkMovStatement(
-                        new IR::PathExpression("metadata"), new IR::Constant(error_id)));
+            if (condition->expression->is<IR::BoolLiteral>()) {
```
Comment (mbudiu-vmw) [inline backends/dpdk/dpdkHelpers.cpp:631]: are these two branches of the if swapped?

**Review by mbudiu-vmw — CHANGES_REQUESTED** [review]

File: backends/dpdk/dpdkHelpers.cpp (hunk 1 above)
Comment (usha1830) [reply backends/dpdk/dpdkHelpers.cpp]: No. This code is for the case where both the  operands of conditional jump are constant. It moves the first operand to a new temporary variable.
Actually, I think I can remove the conditional branch altogether for this case.

File: backends/dpdk/dpdkHelpers.h [hunk 2]
```diff
@@ -127,12 +127,13 @@ class ConvertStatementToDpdk : public Inspector {
     P4::ReferenceMap *refmap;
     DpdkProgramStructure *structure;
     const IR::P4Parser *parser = nullptr;
+    IR::Type_Struct *metadataStruct;
```
[the remark this answers is not in the document]
Comment (usha1830) [reply backends/dpdk/dpdkHelpers.h]: This structure is not needed anymore and is removed.
Agree with you on keeping the interface clean. Will keep in mind in future.

File: backends/dpdk/spec.cpp [hunk 3]
```diff
@@ -545,9 +545,7 @@ std::ostream &IR::DpdkGetChecksumStatement::toSpec(std::ostream &out) const {
 }
 
 std::ostream &IR::DpdkCastStatement::toSpec(std::ostream &out) const {
-    out << "cast "
-        << " " << DPDK::toStr(dst) << " " << DPDK::toStr(type) << " "
-        << DPDK::toStr(src);
+    out << "mov " << " " << DPDK::toStr(dst) << " " << DPDK::toStr(src);
```
Comment (hanw) [inline backends/dpdk/spec.cpp:548]: nit: there is an extra space after 'mov'

**Review by hanw — APPROVED** [review]

File: backends/dpdk/spec.cpp (hunk 3 above)
Comment (usha1830) [reply backends/dpdk/spec.cpp]: @hanw Thanks for the review.
Removed extra space and updated reference outputs.

File: backends/dpdk/dpdkHelpers.cpp (hunk 1 above)
Comment (usha1830) [reply backends/dpdk/dpdkHelpers.cpp]: @mbudiu-vmw Thanks for the review.. 
Could you please have a look at the updated code and see if it looks fine now?

**Review by mbudiu-vmw — APPROVED** [review]

**Pull request merged**